src/Main.elm

```diff
@@ -1,71 +1,392 @@
+            )
+
+        NoOp ->
+            ( model, Cmd.none )
+
+        EditTodo theTodo title ->
+            ( { model
+                | todos =
+                    List.map
+                        (\todo ->
+                            if todo.id == theTodo.id then
+                                { todo | title = title }
+
+                            else
+                                todo
+                        )
+                        model.todos
+              }
+            , Cmd.none
+            )
 
 
 
 -- VIEW
 
 
-view : Model -> Html msg
+view : Model -> Html Msg
 view model =
-    ul []
-        (viewTodos model.todos)
+    div
+        [ css
+            [ maxWidth (px 800)
+            , margin auto
+            , position relative
+            ]
+        ]
+        [ viewHeader model
+        , ul []
+            (viewTodos model)
+        ]
+
+
+onKeyDown : (Int -> msg) -> Attribute msg
+onKeyDown tagger =
+    on "keydown" (Json.map tagger keyCode)
+
 
+viewHeader : Model -> Html Msg
+viewHeader model =
+    header
+        [ css
+            [ textAlign center
+            ]
+        ]
+        [ h1 [] [ text "Todos" ]
+        , input
+            [ css
+                [ Css.width (pct 75)
+                ]
+            , value model.newTodoTitle
+            , onInput ChangeInput
+            , onKeyDown (KeyDown OnAdd)
+            , placeholder "What needs to be done"
+            ]
+            []
+        , button [ onClick AddTodo ] [ text "Add" ]
+        , div []
+            [ input
+                [ onCheck HideCompleted
+                , type_ "checkbox"
+                , Html.checked model.hideCompleted
+                ]
+                []
+            , label [] [ text "Hide completed" ]
+            ]
+        ]
+
+
+viewTodo : Model -> Todo -> Html Msg
+viewTodo model todo =
+    li
+        [ css
+            [ listStyleType none
+            , backgroundColor (rgb 240 240 240)
+            , padding (px 10)
+            , Css.nthChild "even"
+                [ backgroundColor (rgb 220 220 220)
+                ]
+            , if todo.completed then
+                textDecoration lineThrough
 
-viewTodo : Todo -> Html msg
-viewTodo todo =
-    li []
-        [ h1 [] [ text todo.title ]
+              else
+                textDecoration none
+            ]
         ]
+        [ div []
+            [ input
+                [ type_ "checkbox"
+                , Html.checked todo.completed
+                , onCheck (ChangeCompleted todo)
+                ]
+                []
+            , if model.todoBeingEdited == todo.id then
+                input
+                    [ id "todo-edit-input"
+                    , value todo.title
+                    , onBlur (EndEditing todo)
+                    , onInput (EditTodo todo)
+                    , onKeyDown (KeyDown OnRename)
+                    ]
+                    []
+
+              else
```

Comment:
Consider extracting this views into different functions `editTodoView` and `todoView` for example